flytekit/annotated/reference_task.py

```diff
@@ -46,49 +24,49 @@ def __init__(
         task_function: Callable,
         ignored_metadata: _task_model.TaskMetadata,
         *args,
-        task_type="reference-task",
         **kwargs,
     ):
-        self._reference = task_config
-        self._native_interface = transform_signature_to_interface(inspect.signature(task_function))
+        """
+        The signature for this function is different than the other Reference Entities because it's how the task
+        plugins will call it.
+        """
+        interface = transform_signature_to_interface(inspect.signature(task_function))
         super().__init__(
-            task_type=task_type,
-            name=task_config._id._name,
-            interface=self._native_interface,
-            metadata=metadata(),
-            *args,
-            **kwargs,
+            _identifier_model.ResourceType.TASK,
+            task_config.id.project,
+            task_config.id.domain,
+            task_config.id.name,
+            task_config.id.version,
+            interface.inputs,
+            interface.outputs,
         )
-        self._task_function = task_function
 
-    def execute(self, **kwargs) -> Any:
-        raise Exception("Remote reference tasks cannot be run locally. You must mock this out.")
+        self._registerable_entity: Optional[SdkTask] = None
 
-    @property
-    def reference(self) -> TaskReference:
-        return self._reference
+    @classmethod
+    def create_from_get_entity(
```

Comment:
thats a weird method name - with is `*_get_entity` Also dont understand the usecase for this method

Reply:
The ReferenceTask class is hooked up to the task decorator (the bit that picks the constructor to use based on the config class type).  That function calls the task in a specific way

```
task_instance = TaskPlugins.find_pythontask_plugin(type(task_config))(
       task_config, fn, _metadata, container_image=container_image, *args, **kwargs
)
```

This means that the constructor for ReferenceTask is different from launch plans and workflows, so I wanted another object creation function that has the same signature.

Reply:
and we have to use `@task` annotation for this? Do you think a different one makes sense, especially because `container_image` is really weird?

Reply:
Nope, don't have to use @task for it at all.  I can rip that out completely so that users have to use the `get_reference_entity` function for tasks.  Just thought the empty function body style looked kinda nice but if it's just getting in the way, let's get rid of it.